pkg/kube/inject/webhook.go

```diff
@@ -833,14 +833,40 @@ func (wh *Webhook) serveInject(w http.ResponseWriter, r *http.Request) {
 }
 
 // parseInjectEnvs parse new envs from inject url path
-// follow format: /inject/k1/v1/k2/v2, any kv order works
-// eg. "/inject/cluster/cluster1", "/inject/net/network1/cluster/cluster1"
+// follow format: /inject/k1/v1/k2/v2 when values do not contain slashes,
+// follow format: /inject/:ENV:net=network1:ENV:cluster=cluster1:ENV:rootpage=/foo/bar
```

Comment:
just hope we never need an `=` in the env var..

Reply:
Actually `=` works too because it's not a valid char in a variable name, and the code is only looking for the first one. Any following ones will just be included as part of the value.

Reply:
@howardjohn @frankbu , that is right. If a value contains =, it will be as is. It just cannot be part of the variable name. For example, /inject/:ENV:net=network1:ENV:cluster=cluster1:ENV:rootpage=/foo?&abc=123
This rootpage variable which contains query parameters and will work with no issues. The value will be /foo?abc=123